### ISSUE TITLE
Refine Oxygen code templates

### ISSUE THREAD

**Issue by nathangibson** [usaybia/usaybia-data#55]
@lunadine I've created a couple code templates for Oxygen that you can try out. Sorry it took so long - I ran into some issues that were more complicated than I expected!

If these don't work, make sure you have Oxygen 21.1 or higher. 

**To install these**
 1. Save this file on your computer: https://raw.githubusercontent.com/usaybia/usaybia-tools/master/tools/oxygen/code-templates.xml
 1. In Oxygen go to Einstellungen > Editor > Autovervollständigung > Code-Vorlagen
 1. Click Importieren and select the downloaded file.

**How to use these**
There are two templates. One is to insert the factoid div without the specific content, e.g., of a relation. The other is to insert the content of a relation between two people.
 1. **In your factoid document**, insert a space between 2 factoid divs. Then press Ctrl-Shift-F (for "factoid"). 
 1. Type the factoid subtype in the pop-up (e.g., "relation").
 1. A factoid will be inserted with the following values.
    1. `@subtype` whatever you put in the pop-up
    1. `@xml:id="factoid-156"` with the highest number ID you've used so far in the doc + 1
    1. `resp="#nloehr"` using the resp of the previous factoid
    1. ` <idno type="URI">https://usaybia.net/factoid/14-25-156</idno>` again using the highest number + 1
    1. The same `<bibl>` as the one you used in the last factoid -- but you should update this each time you move to a new subsection. 
    1. Between idno and bibl will be a place for you to insert the content of your factoid, e.g., with a listRelation
 1. **In the text where you've tagged the names**, etc. (e.g., lhom-14) select the two persNames you want to have in the relation (it's fine if there's also text in between).
 ~~1. **In the text where you've tagged the names**, etc. (e.g., lhom-14) select an element that _contains_ the persNames you want to have in the relation. (Important note: you can't select multiple "sibling" nodes, it has to be a single element that contains the persNames somewhere inside it, such as a `<p>`. It will use the first 2 persNames in that element. If you want to create a relation for other persons, one way to do it is to select the two you need, press Ctrl-E and type "span" to surround them with a span element which you can delete later. Then you can select this. A quick way to select an entire element is to triple-click.)~~
 1. Press Ctrl-Shift-R on the selection.
 1. Type the kind of relation for the `@ref` (e.g., `usaybia:colleague-of`)
 1. It will add a `<listRelation>` after your selection. 
 1. Cut and paste this listRelation into your factoid document, in the middle of the factoid you just created. 
 1. The listRelation will have 
    1. The `@ref` you just typed
    1. `@mutual` containing the URIs of the persons you selected. Sometimes you'll need to change this to `@active` and `@passive`
    1. A `<desc>` containing empty persNames tagged with the URIs. Replace the commented parts with the correct names and a phrase describing the relationships, so that it reads like a sentence. (E.g., "Ibn Ridwan had enmity for Ibn Butlan.") 

I know this sounds complicated but once you have the templates installed and you remember the keyboard shortcuts it should make it really fast. 

Can you please test these and let me know what we should tweak? Once these are working well we can add others, for other types of factoids or even specific types of relations.

<img width="1192" alt="Bildschirmfoto 2020-07-22 um 15 30 14" src="https://user-images.githubusercontent.com/3083452/88182278-49fc7600-cc30-11ea-828b-dce9af6be348.png">

**Comment by nathangibson** [comment]
@lunadine If you can, it would be good to install the new version of the templates I just uploaded: https://github.com/usaybia/usaybia-data/issues/55. I figured out how you can select just the persNames you need without having to select or insert a parent element.

**Comment by lunadine** [comment]
Dear Nathan, 
thanks so much for this perfectly detailed explanation and the templates, this is perfect! I installed Oxygen 22 last week and I will test the templates on the weekend. Sorry I have not done it yet, but we had some meetings and I accidentally left the laptop on which I work with oxygen at work.  I will let you know if I face any problems and I will also be in the institute on monday.
See you in few min.

**Comment by nathangibson** [comment]
No problem. You should be able to install a trial version of Oxygen 22 on any computer, so if you're able that would be helpful for today's meeting.

**Comment by nathangibson** [comment]
I've updated the code templates again at https://raw.githubusercontent.com/usaybia/usaybia-tools/master/tools/oxygen/code-templates.xml. You should be able to just import the new file. 

They now include the most common types of factoid content:
 - mutual relation
 - active/passive relation 
 - active/passive relation with person 1 and 2 reversed (caution!)
 - event
 - birthDate/deathDate
 - birthPlace/deathPlace
 - occupation

Each has a keyboard shortcut. You can also get to them with Ctrl-Shift-Space (Windows?) and then selecting one. 

Let me know what needs to be adjusted. 

I figured out it is possible to automatically insert the person/place name from the person/place record, but to do this you'll have to have the full Github repo downloaded and be working on a file inside it. Your branch doesn't yet have person or place records -- I can merge them in but first you'll need to synchronize your branch with the local changes you've made on your computer. So let me know when you've synchronized and then I'll merge stuff in and add the automatic person/place name feature to the code templates. :-)

**Comment by lunadine** [comment]
Hello Nathan,
thanks so much! I imported the templates on my other laptop now. I am just not sure if I correctly synced my branch,
there is a conflict between the branches (which makes sense as I changed one file) so I did not want to pull origin. I sent a pull request. I hope this was correct, I googled about the synchronization as I was uncertain, but the explanation I found suggested a pull request. Talk to you later :)
